x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/validation/SourceDestValidator.java

```diff
@@ -0,0 +1,513 @@
+        new DestinationInSourceValidation(),
+        new DestinationSingleIndexValidation()
+    );
+
+    /**
+     * Validate dest request.
+     *
+     * This runs a couple of simple validations at request time, to be executed from a {@link ActionRequest}}
+     * implementation.
+     *
+     * Note: Source can not be validated at request time as it might contain expressions.
+     *
+     * @param validationException an ActionRequestValidationException for collection validation problem, can be null
+     * @param dest destination index, null if validation shall be skipped
+     */
+    public static ActionRequestValidationException validateRequest(
+        @Nullable ActionRequestValidationException validationException,
+        @Nullable String dest
+    ) {
+        try {
+            if (dest != null) {
+                validateIndexOrAliasName(dest, InvalidIndexNameException::new);
+                if (dest.toLowerCase(Locale.ROOT).equals(dest) == false) {
+                    validationException = addValidationError(getMessage(DEST_LOWERCASE, dest), validationException);
+                }
+            }
+        } catch (InvalidIndexNameException ex) {
+            validationException = addValidationError(ex.getMessage(), validationException);
+        }
+
+        return validationException;
+    }
+
+    /**
+     * Validates source and dest indices for preview.
+     *
+     * @param clusterState The current ClusterState
+     * @param indexNameExpressionResolver A valid IndexNameExpressionResolver object
+     * @param remoteClusterService A valid RemoteClusterService object
+     * @param remoteClusterLicenseChecker A RemoteClusterLicenseChecker or null if CCS is disabled
+     * @param source an array of source indices
+     * @param dest destination index
+     * @param nodeName the name of this node
+     * @param license the license of the feature validated for
+     * @return ValidationException instance containing all validation errors or null if validation found no problems.
+     */
+    public static ValidationException validateForPreview(
+        ClusterState clusterState,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        RemoteClusterService remoteClusterService,
+        RemoteClusterLicenseChecker remoteClusterLicenseChecker,
+        String[] source,
+        String dest,
+        String nodeName,
+        String license
+    ) {
+        Context context = new Context(
+            clusterState,
+            indexNameExpressionResolver,
+            remoteClusterService,
+            remoteClusterLicenseChecker,
+            source,
+            dest,
+            nodeName,
+            license
+        );
+
+        // validation exception might gets thrown if validation stops early
+        try {
+            for (SourceDestValidation validation : PREVIEW_VALIDATIONS) {
+                validation.validate(context);
+            }
+        } catch (ValidationException e) {}
```

Comment:
I don't think anything throws this? If something does, we should not `> /dev/null` this. Add it to the overall validation error collection.